src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -333,10 +332,8 @@ export const useBalanceChartData: UseBalanceChartData = args => {
     // data prep
     const noDeviceId = isNil(walletInfo?.deviceId)
     const noAssetIds = !assetIds.length
-    const noTxs = !txs.length
-    const noBalances = isEmpty(balances)
     const noPriceHistory = priceHistoryDataLoading
-    if (noDeviceId || noAssetIds || noTxs || noBalances || noPriceHistory) {
```

Comment:
hi @jdambron,
what was the issue exactly? 

I might be wrong but It seems like this change introduces a new issue:
checks for `txs` & `balances` make sense as these values passed to `bucketTxs ` and `makeBuckets` later

https://github.com/shapeshift/web/blob/755853c36c3a107320a2891411e28fb85393cf66/src/hooks/useBalanceChartData/useBalanceChartData.ts#L105
`makeBuckets` will throw an error if `balances` is empty

---

it would be great to add a couple of tests for `useBalanceChartData`

Reply:
Hi @daodev877.
From my understanding, the issue was that we return `setBalanceChartDataLoading(true)` in this case, meaning that we stay in a loading state endlessly if we have no balances / no transactions.
I may be wrong.

Reply:
Hey guys

You're both pretty close here

The way I would address this is pull in the portfolio loaded selector that checks if we have accounts, and also if txs has finished loading. 

This is kind of tricky as we can't *really* tell when txs have finished loading, we just debounce the web socket.

I probably wouldn't remove the balanced or txs check completely as then the chart will probably furiously render as each tx comes in. 

I'll take a look locally when I'm at a computer that this doesn't impact perf badly with an account with lots of txs.

Reply:
ok, i pulled this locally and had a play, it actually performs fine - no crazy renders and works for both empty wallets and wallets with significant tx history. 

it seems the debounce is enough to stop the furious rendering, and @daodev877 the `isEmpty(balances)` check seemed redundant, and this works as expected.

a lot of those guard conditions were left over from before we did a big state refactor, it's possible more could go away, but for now this is great.